pkg/admin/service/instance.go

```diff
@@ -43,3 +44,33 @@ func SearchInstances(rt core_runtime.Runtime, req *model.SearchInstanceReq) ([]*
 
 	return res, &dataplaneList.Pagination, nil
 }
+
+func GetInstanceDetail(rt core_runtime.Runtime, req *model.InstanceDetailReq) ([]*model.InstanceDetailResp, error) {
+	manager := rt.ResourceManager()
+	dataplaneList := &mesh.DataplaneResourceList{}
+
+	if err := manager.List(rt.AppContext(), dataplaneList, store.ListByNameContains(req.InstanceName)); err != nil {
+		return nil, err
+	}
+
+	instMap := make(map[string]*model.InstanceDetail)
+	for _, dataplane := range dataplaneList.Items {
+		instName := dataplane.Meta.GetLabels()[mesh_proto.InstanceTag]
+		var instanceDetail *model.InstanceDetail
+		if _, ok := instMap[instName]; ok {
+			instanceDetail = instMap[instName]
+		} else {
+			instanceDetail = model.NewInstanceDetail()
+		}
+		instanceDetail.Merge(dataplane) //convert dataplane info to instance detail
+		instMap[instName] = instanceDetail
```

Comment:
Should we only set the map `instMap[instName] = instanceDetail` in the `else` branch above?

Reply:
The logic is ok in my opinion, considering the need for merging different information from the same instance. The explanation of the logic is shown in comments in this commit:  https://github.com/apache/dubbo-kubernetes/pull/286/commits/e82fa817dc25b9cea2e50a2ecff9e071d8a33a20

Reply:
I see